ssz-rs/src/merkleization/mod.rs

```diff
@@ -265,27 +283,36 @@ mod tests {
         debug_assert!(chunks.len() % BYTES_PER_CHUNK == 0);
         debug_assert!(leaf_count.next_power_of_two() == leaf_count);
 
+        // checked subtraction is unnecessary, as leaf_count != 0 (0.next_power_of_two() == 1); qed
         let node_count = 2 * leaf_count - 1;
+        // checked subtraction is unnecessary, as node_count >= leaf_count; qed
         let interior_count = node_count - leaf_count;
         let leaf_start = interior_count * BYTES_PER_CHUNK;
 
         let mut hasher = Sha256::new();
         let mut buffer = vec![0u8; node_count * BYTES_PER_CHUNK];
         buffer[leaf_start..leaf_start + chunks.len()].copy_from_slice(chunks);
-        let zero_chunk = [0u8; 32];
+        let zero_chunk = [0u8; BYTES_PER_CHUNK];
         for i in chunks.len()..leaf_count {
             let start = leaf_start + (i * BYTES_PER_CHUNK);
             let end = leaf_start + (i + 1) * BYTES_PER_CHUNK;
             buffer[start..end].copy_from_slice(&zero_chunk);
         }
 
         for i in (1..node_count).rev().step_by(2) {
+            // checked subtraction is unnecessary, as i >= 1; qed
             let parent_index = (i - 1) / 2;
             let focus = &mut buffer[parent_index * BYTES_PER_CHUNK..(i + 1) * BYTES_PER_CHUNK];
+            // checked subtraction is unnecessary:
+            // focus.len() = (i + 1 - parent_index) * BYTES_PER_CHUNK
+            //             = ((2*i + 2 - i + 1) / 2) * BYTES_PER_CHUNK
+            //             = ((i + 3) / 2) * BYTES_PER_CHUNK
+            // and
+            // i >= 1
+            // so focus.len() >= 2 * BYTES_PER_CHUNK; qed
             let children_index = focus.len() - 2 * BYTES_PER_CHUNK;
             let (parent, children) = focus.split_at_mut(children_index);
-            let left = &children[0..BYTES_PER_CHUNK];
-            let right = &children[BYTES_PER_CHUNK..2 * BYTES_PER_CHUNK];
+            let (left, right) = children.split_at(BYTES_PER_CHUNK);
```

Comment:
let's verify that `children.len() == 32` _always_ and then I agree this is better

Reply:
We should have `children.len() == 64` (`2 * BYTES_PER_CHUNK`) always because:
- Above `(parent, children) = focus.split_at_mut(children_index) == focus.split_at_mut(focus.len() - 2 * BYTES_PER_CHUNK)`, so `children.len() <= 2 * BYTES_PER_CHUNK`.
and
- We know that `focus.len() >= 2 * BYTES_PER_CHUNK`, so we must have `children.len() >= 2 * BYTES_PER_CHUNK`.

Reply:
yep I see, thanks!